src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -57,6 +57,34 @@ function migrateIndexPattern(doc) {
   doc.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(searchSource);
 }
 
+function removeDateHistogramTimeZones(doc) {
+  const newDoc = cloneDeep(doc);
```

Comment:
nit: Not a huge thing, but you don't necessarily need to clone every doc that comes in since you are only going to be updating `attributes.visState` on date histograms. You could return early if the doc is not identified as needing to be changed, and only clone / reassign the `visState` on docs that you know you need to touch.